src/components/Layout/Warning.tsx

```diff
@@ -1,18 +1,51 @@
+import React, { useContext, useCallback, useState } from 'react'
 import style from "./Warning.module.scss"
-import layout from "../../styles/Layout.module.scss"
 import cs from "classnames"
+import { TopBannerContext, ITopBannerContext } from '../../context/TopBanner'
 import { FunctionComponent } from "react"
 import Link from "next/link"
 
 export const Warning: FunctionComponent = ({ children }) => {
+  const [message, setMessage] = useState<string|null>(null);
+  const { 
+    history, 
+    addMessage 
+  } = useContext<ITopBannerContext>(TopBannerContext)
+
+  const isOpen = !message || !history.find(m => m.text === message); 
+
+  const handleClose = () => {
+    if (isOpen && message) {
+      addMessage({
+	text: message,
+	from: Date.now(), 
+      }) 
+    }
+  }
+
+  const messageRef = useCallback(node => {
+    if (node) setMessage(node.textContent)
+  }, [children])
```

Comment:
although currently we set the message via `.env` we might want to design more complex banners. therefore the `textContent` is extracted and this is what is stored in the localStorage and checked for.